client/extensions/woocommerce/state/data-layer/ui/products/index.js

```diff
@@ -2,23 +2,49 @@
  * External dependencies
  */
 import { translate } from 'i18n-calypso';
+import { find } from 'lodash';
 
 /**
  * Internal dependencies
  */
 // TODO: Remove this when product edits have siteIds.
 import { getSelectedSiteId } from 'state/ui/selectors';
+import { editProductRemoveCategory } from 'woocommerce/state/ui/products/actions';
 import { getAllProductEdits } from 'woocommerce/state/ui/products/selectors';
+import { getAllProductCategoryEdits } from 'woocommerce/state/ui/product-categories/selectors';
 import { createProduct } from 'woocommerce/state/sites/products/actions';
 import {
 	actionListCreate,
 	actionListStepSuccess,
 	actionListStepFailure
 } from 'woocommerce/state/action-list/actions';
 import {
+	WOOCOMMERCE_PRODUCT_CATEGORY_EDIT,
 	WOOCOMMERCE_PRODUCT_ACTION_LIST_CREATE,
 } from 'woocommerce/state/action-types';
 
+export function handleProductCategoryEdit( { dispatch, getState }, action ) {
+	const rootState = getState();
+	const { siteId, category, data } = action;
+
+	if ( null === data ) {
+		// It's removing a category from edits.
+		const categoryCreates = getAllProductCategoryEdits( rootState, siteId ).creates;
+		if ( find( categoryCreates, c => category.id === c.id ) ) {
```

Comment:
Consider using the simpler `find( categoryCreates, { id: category.id } )` instead